File: src/main/kotlin/net/yttravelers/spotshare/entities/Route.kt
package net.yttravelers.spotshare.entities

import javax.persistence.*

@Entity
@Table(name = "route")
class Route {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var routeId: Int? = null

    @Column(name = "name")
    var routeName: String? = null

    @OneToMany(fetch = FetchType.EAGER, cascade= [CascadeType.ALL],  mappedBy = "route")
    var routeDetails: MutableList<RouteDetail> = mutableListOf()
}

File: src/main/angular/src/app/shared/model/route.ts
import { IRouteDetail } from './route-detail';

/**
 * ルート
 */
export interface IRoute {
  /** ルートID */
  routeId?: number;
  /** ルート名 */
  routeName?: string;
  /** ルート詳細 */
  routeDetails?: IRouteDetail[];
}
